app/LoadingApp/LoadingApp.tsx

```diff
@@ -489,8 +489,18 @@ export class LoadingAppClass extends Component<LoadingAppClassProps, LoadingAppC
                 //console.log(walletKindStr);
                 try {
                   const walletKindJSON: RPCWalletKindType = await JSON.parse(walletKindStr);
+                  console.log(walletKindJSON);
+                  // there are 4 kinds:
+                  // 1. seed
+                  // 2. USK
+                  // 3. UFVK - watch-only wallet
+                  // 4. No keys - watch-only wallet (possibly an error)
                   this.setState({
-                    readOnly: walletKindJSON.kind === RPCWalletKindEnum.Seeded ? false : true,
+                    readOnly:
+                      walletKindJSON.kind === RPCWalletKindEnum.LoadedFromUnifiedFullViewingKey ||
+                      walletKindJSON.kind === RPCWalletKindEnum.NoKeysFound
```

Comment:
@Oscar-Pepper What do you think about this? `NoKeysFound` should be an error, and the app doesn't load the wallet, or It's OK if the App load it as a `Watch/read-only` wallet. ???

Reply:
I assume WalletKind is never called in a state where there is no keys. Without keys the wallet can't sync or create addresses so maybe zingomobile should error? I don't see an issue with treating it as. UfVK with no view keys though if that's easier

Reply:
Since it is an impossible use case, I think can be more interesting to load it as a Watch/Read-only wallet. IMO.